### PR TITLE
build/ops: fix build fail related to PYTHON_EXECUTABLE variable

### PR DESCRIPTION
**Pull request by IlsooByun**
tracker: https://tracker.ceph.com/issues/41676

Signed-off-by: Ilsoo Byun <ilsoobyun@linecorp.com>

## Checklist
- [x] References tracker ticket
- [x] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`

</details>

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -740,6 +740,7 @@ install(FILES
   sample.ceph.conf
   DESTINATION ${CMAKE_INSTALL_DOCDIR})
 
+set(PYTHON_EXECUTABLE "${PYTHON_EXECUTABLE}" STRING)
```
Comment (tchaikov) [inline src/CMakeLists.txt]: hi @IlsooByun, thanks for fixing this issue. i think a better fix would be 

1. s/PYTHON_EXECUTABLE/MGR_PYTHON_EXECUTABLE/ in https://github.com/ceph/ceph/blob/49addda5707f06b90746bee96b82c4d77a23951b/src/include/config-h.in.cmake#L319. as per https://cmake.org/cmake/help/v3.2/command/configure_file.html?highlight=cmakedefine#configure-file what variable quoted by `@` is does not matters. so the existing .in file is misleading, probably we can make it more consistent? in this case, it should represent `MGR_PYTHON_EXECUTABLE`.
1. s/PYTHON_EXECUTABLE/MGR_PYTHON_EXECUTABLE/ in https://github.com/ceph/ceph/blob/49addda5707f06b90746bee96b82c4d77a23951b/src/mgr/PyModuleRegistry.cc#L44 and https://github.com/ceph/ceph/blob/49addda5707f06b90746bee96b82c4d77a23951b/src/mgr/PyModuleRegistry.cc#L47 as well

what do you think?

Comment (IlsooByun) [reply src/CMakeLists.txt]: I agree with you. I tested in a similar fashion but chose this approach to reduce source code changes. I'll change as you said and push it again.

**Comment by tchaikov** [comment]
also please add "Fixes: https://tracker.ceph.com/issues/41676" right before your "Signed-off-by" line in your commit message.

File: src/CMakeLists.txt (hunk 1 above)
Comment (tchaikov) [reply src/CMakeLists.txt]: @IlsooByun probably we don't think change with the rest of the changes in this PR? actually, i still cannot understand why it addresses the failure you described in the tracker ticket..

**Comment by xiexingguo** [comment]
@tchaikov Sorry for OT:
I think the qa run of http://pulpito.ceph.com/kchai-2019-09-06_03:33:37-rados-wip-kefu-testing-2019-09-02-1658-distro-basic-smithi/ is broken(I try to ssh to the test machines but got EPERM :( ).
Would you mind taking a look? Thanks!

**Comment by tchaikov** [comment]
@xiexingguo please feel free to ping me over IRC \#ceph-devel if anything urgent or drop me an email if anything urgent.

File: src/CMakeLists.txt (hunk 1 above)
Comment (IlsooByun) [reply src/CMakeLists.txt]: **cmakedefine** checks if the variable is already set. In this case, PYTHON_EXECUTABLE is not set, **cmakedefine** generates '/* undef PYTHON_EXECUTABLE*/'. So I had to set any value to PYTHON_EXECUTABLE to compile the code. Considering the case where PYTHON_EXECUTABLE has already been used elsewhere, this variable was set to its own value.

Comment (IlsooByun) [reply src/CMakeLists.txt]: I'm sorry for the confusion. I'll remove that line.

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2019-09-08_13:41:17-rados-wip-kefu-testing-2019-09-07-0159-distro-basic-smithi/
- http://pulpito.ceph.com/kchai-2019-09-07_14:44:52-rados-wip-kefu-testing-2019-09-07-0159-distro-basic-mira/

**Pull request merged**